### PR TITLE
Always link the c_libraries in the configurator build command

### PR DESCRIPTION
**Pull request by MisterDA**
With mingw-w64 on Windows, the C libraries (usually `-lws2_32
-lversion`) are not automatically linked, contrary to some Linux
systems. In dune-configurator, the cflags are always picked, the link
flags should too. Otherwise, they must be added manually in the
discover script.

Signed-off-by: Antonin Décimo <antonin@tarides.com>

### CODE REVIEW

**Comment by MisterDA** [comment]
I think the PR is ready but I can't figure out how to make dune-configurator use the secondary compiler. I need `String.split_on_char` but that's 4.04. Should I provide a replacement for that function instead?

**Comment by rgrinberg** [comment]
> I think the PR is ready but I can't figure out how to make dune-configurator use the secondary compiler

We can't have dune-configurator use the secondary compiler.

>  I need String.split_on_char but that's 4.04. Should I provide a replacement for that function instead?

Unfortunately, yes. If you search through the gist history you might be able to revive one of our old implementations.

**Comment by MisterDA** [comment]
Thanks for the tips, I ditched the commit trying to upgrade configurator to OCaml 4.04 and embedded a version of `String.split_on_char` I found in the history as you suggested.

File: otherlibs/configurator/src/v1.ml [hunk 1]
```diff
@@ -308,18 +309,34 @@ let read_dot_dune_configurator_file ~build_dir =
     in
     { ocamlc; vars }
 
+(* Needed for OCaml 4.03 compat, String.split_on_char was introduced in 4.04. *)
+let split_on_char s ~sep =
```
Comment (rgrinberg) [inline otherlibs/configurator/src/v1.ml:313]: Actually, this isn't unnecessary after all. `import.ml` already contains `split`. Sorry for the trouble.

**Comment by rgrinberg** [comment]
Is it possible to add a test for this?

File: otherlibs/configurator/src/v1.ml (hunk 1 above)
Comment (MisterDA) [reply otherlibs/configurator/src/v1.ml]: I've used `Configurator.V1.Flags.extract_blank_separated_words`.

**Comment by MisterDA** [comment]
> Is it possible to add a test for this?

I've added a test that show that before the patch the library flags are not picked by configurator and the compilation fails, and after the patch compilation succeeds.

**Comment by rgrinberg** [comment]
Thank you for  your contribution.

**Pull request merged**